content/automate/ManageIQ/System/Notification/Email.class/__methods__/miq_provision_complete.rb

```diff
@@ -0,0 +1,76 @@
+#
+# Description: This method sends an e-mail when the following event is raised:
+# Events: vm_provisioned
+#
+# Model Notes:
+# 1. to_email_address - used to specify an email address in the case where the
+#    vm's owner does not have an  email address. To specify more than one email
+#    address separate email address with commas. (I.e. admin@example.com,user@example.com)
+# 2. from_email_address - used to specify an email address in the event the
+#    requester replies to the email
+# 3. signature - used to stamp the email with a custom signature
+#
+
+module ManageIQ
+  module Automate
+    module System
+      module Notification
+        module Email
+          class MiqProvisionComplete
+            def initialize(handle = $evm)
+              @handle = handle
+            end
+
+            def main
+              @handle.log("info", "Starting miq_provision_complete")
+              check_email
+              @handle.log("info", "Ending miq_provision_complete")
+            end
+
+            private
+
+            def check_email
+              prov = @handle.root['miq_provision']
```

Comment:
@billfitzgerald0120 
We would have to check if miq_provision exists in the root object?

I was thinking writing small functions to get these and have them raise errors if the required objects are not in the objects.

Reply:
@billfitzgerald0120 we also need specs we already have some template specs that we could use to test the presence of all variables and a precanned syntax for the body.